src/classes/storage.ts

```diff
@@ -84,4 +100,155 @@ export class Storage {
+        return place;
+      }
+    });
+  }
+
+  // -----------------------------------------------------------------------------
+  // ----------------------------- PRIVATE METHODS -------------------------------
+  // -----------------------------------------------------------------------------
+
+  /**
+   * @description Converts the Ground places to an array list manipulable.
+   * @returns {AutoComplete[]}
+   */
+  private parseGroundPlaces(): AutoComplete[] {
+    return Object.entries(this.groundPlaces).map(
+      ([gpuid, place]: [StopGroupGpuid | StopClusterGpuid, StopGroup | StopCluster]) => ({
+        gpuid,
+        ...place,
+      }),
+    );
+  }
+
+  /**
+   * @description Filter Ground places by group type.
+   * @param {AutoComplete[]} groundPlaces - Ground places list to be filtered.
+   * @param {boolean} filterByGroup - Filter or not by group type.
+   * @returns {AutoComplete[]}
+   */
+  private filterGroundPlacesByGroup(groundPlaces: AutoComplete[], filterByGroup: boolean): AutoComplete[] {
+    if (filterByGroup) {
+      return groundPlaces.filter((place: AutoComplete) => place.type === GroundPlaceType.GROUP);
+    }
+
+    return [];
+  }
+
+  /**
+   * @description Filter Ground places by cluster type.
+   * @param {AutoComplete[]} groundPlaces - Ground places list to be filtered.
+   * @param {boolean} filterByCluster - Filter or not by cluster type.
+   * @returns {AutoComplete[]}
+   */
+  private filterGroundPlacesByCluster(groundPlaces: AutoComplete[], filterByCluster: boolean): AutoComplete[] {
+    if (filterByCluster) {
+      return groundPlaces.filter((place: AutoComplete) => place.type === GroundPlaceType.CLUSTER);
+    }
+
+    return [];
+  }
+
+  /**
+   * @description Filter Ground places by Segment Provider in it.
+   * @param {AutoComplete[]} groundPlaces - Ground places list to be filtered.
+   * @param {boolean} filterBySegmentProvider - Filter or not by segment provider in it.
+   * @returns {AutoComplete[]}
+   */
+  private filterGroundPlacesBySegmentProvider(
+    groundPlaces: AutoComplete[],
+    filterBySegmentProvider: boolean,
+  ): AutoComplete[] {
+    if (filterBySegmentProvider) {
+      // Since StopGroups and StopClusters do not share the same structures
+      // We have to search the StopGroups from the StopCluster in order to find the potential segmentProviderStop.
+      return groundPlaces.filter((place: AutoComplete) => {
+        /* Check inside StopCluster through its StopGroup(s) children(s) */
+        if (place.type === GroundPlaceType.CLUSTER) {
+          const isSegmentProviderExist = place.childs.find(
+            (stopGroupGpuid) => this.getStopGroupByGpuid(stopGroupGpuid).childs.length,
+          );
+
+          if (isSegmentProviderExist) return place;
+          /* Check inside StopGroup through its children(s) */
+        } else if (place.type === GroundPlaceType.GROUP && place.childs.length) {
```

Comment:
i dont understand the `place.childs.length` condition. if a stop group has children the its in the filter ?